### PR TITLE
K8SPSMDB-226 PSMDB database object helm chart

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

File: charts/psmdb-db/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+| `backup.storages`   | Local/remote backup storages settings   | `{}` |
+| `backup.tasks`   | Backup working schedule   | `{}` |
+| `users`   | PSMDB essential users   | `{}` |
+
+
+Specify parameters using `--set key=value[,key=value]` argument to `helm install`
+
+## Examples
+
+### Deploy a Cluster without a MySQL Proxy, no backups, no persistent disks
```
Comment (fiowro) [inline charts/psmdb-db/README.md:93]:
```suggestion
### Deploy a Cluster with no MySQL Proxy, no backups, and no persistent volumes
```

File: charts/psmdb-db/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Custom Resource Details
+* <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>
+
+## StatefulSet Details
+* <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>
+
+## Chart Details
+This chart will:
+* deploy a PSMDB database Pods (Custom Resource -> StatefulSet) for the further MongoDB Cluster creation in K8S.
```
Comment (fiowro) [inline charts/psmdb-db/README.md:20]:
```suggestion
* deploy PSMDB database Pods (Custom Resource -> StatefulSet) for the further MongoDB Cluster creation in K8S.
```

**Review by delgod — APPROVED** [review]

File: charts/psmdb-db/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```sh
+helm repo add percona https://percona-lab.github.io/percona-helm-charts/
+helm install my-db percona/psmdb-db --version 0.1.0 --namespace my-namespace
+```
+
+The chart can be customized using the following configurable parameters:
+
+| Parameter                       | Description                                                                   | Default                                   |
+| ------------------------------- | ------------------------------------------------------------------------------| ------------------------------------------|
+| `nameOverride`                  | Set if you the chart name is not desired to be operators name                 | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-db/README.md:34]:
```suggestion
```
It looks like this option is not officially supported

File: charts/psmdb-db/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+| `fullnameOverride`              | By default operator name will contain chart name concatenated with operator name.<br>Set this variable if you want to set the whole bunch of names  | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-db/README.md:35]:
```suggestion
```

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: charts/psmdb-db/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+| `pause`                         | Stop PSMDB Database safely                                                    | `false`                                   |
```
[the remark this answers is not in the document]
Comment (delgod) [reply charts/psmdb-db/README.md:36]: in Operator - yes, https://github.com/percona/percona-server-mongodb-operator/blob/master/pkg/apis/psmdb/v1/psmdb_defaults.go#L202-L206

File: charts/psmdb-db/production-values.yaml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+# Platform type: kubernetes, openshift
+# platform: kubernetes
+
+# Cluster DNS Suffix
+# DNSsuffix: .svc.cluster.local
+
+nameOverride: ""
+fullnameOverride: ""
+
+pause: false
```
[the remark this answers is not in the document]
Comment (delgod) [reply charts/psmdb-db/production-values.yaml:14]: in Operator - yes, https://github.com/percona/percona-server-mongodb-operator/blob/master/pkg/apis/psmdb/v1/psmdb_defaults.go#L202-L206

**Review by delgod — CHANGES_REQUESTED** [review]

File: charts/psmdb-db/production-values.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (tplavcic) [reply charts/psmdb-db/production-values.yaml:14]: in cr.yaml for PSMDB we don't have it and for PXC we have it: https://github.com/percona/percona-xtradb-cluster-operator/blob/master/deploy/cr.yaml#L27
should it be added for PSMDB also in cr.yaml?

[the remark this answers is not in the document]
Comment (delgod) [reply charts/psmdb-db/production-values.yaml:14]: please add

File: charts/psmdb-db/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+appVersion: "1.4.0"
```
Comment (tplavcic) [inline charts/psmdb-db/Chart.yaml:2]: 1.5.0?

File: charts/psmdb-db/production-values.yaml [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+allowUnsafeConfigurations: false
+# updateStrategy: SmartUpdate
+
+image:
+  repository: percona/percona-server-mongodb-operator
```
Comment (tplavcic) [inline charts/psmdb-db/production-values.yaml:19]: repository should be: percona/percona-server-mongodb
tag: 4.2.8-8

File: charts/psmdb-db/production-values.yaml [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+# imagePullSecrets: []
+# runUid: 1001
+secrets: {}
+
+pmm:
+  enabled: false
+  image:
+    repository: percona/percona-server-mongodb-operator
+    tag: 1.4.0-pmm
```
Comment (tplavcic) [inline charts/psmdb-db/production-values.yaml:30]: tag: 1.5.0-pmm

File: charts/psmdb-db/production-values.yaml [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+# Cluster DNS Suffix
+# DNSsuffix: .svc.cluster.local
+
+nameOverride: ""
+fullnameOverride: ""
+
+pause: false
+allowUnsafeConfigurations: false
+# updateStrategy: SmartUpdate
```
Comment (tplavcic) [inline charts/psmdb-db/production-values.yaml:16]: smart update is enabled by default in the official image, should we enable it here also?
and we should probably add more fields (even if they are commented): https://github.com/percona/percona-server-mongodb-operator/blob/master/deploy/cr.yaml#L14-L18

File: charts/psmdb-db/values.yaml [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+# Platform type: kubernetes, openshift
+# platform: kubernetes
+
+# Cluster DNS Suffix
+# DNSsuffix: .svc.cluster.local
+
+pause: false
+allowUnsafeConfigurations: false
+# updateStrategy: SmartUpdate
```
Comment (tplavcic) [inline charts/psmdb-db/values.yaml:13]: more fields are needed for smartupdate so maybe add them commented if it is going to be disabled here: https://github.com/percona/percona-server-mongodb-operator/blob/master/deploy/cr.yaml#L14-L18

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: charts/psmdb-db/Chart.yaml (hunk 7 above)
Comment (cap1984) [reply charts/psmdb-db/Chart.yaml]: done

File: charts/psmdb-db/production-values.yaml (hunk 8 above)
Comment (cap1984) [reply charts/psmdb-db/production-values.yaml]: done

File: charts/psmdb-db/production-values.yaml (hunk 9 above)
Comment (cap1984) [reply charts/psmdb-db/production-values.yaml]: done

File: charts/psmdb-db/production-values.yaml (hunk 10 above)
Comment (cap1984) [reply charts/psmdb-db/production-values.yaml]: done

File: charts/psmdb-db/production-values.yaml [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    #        accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 3Gi
+backup:
+  enabled: true
+  restartOnFailure: true
+  image: 
+    repository: percona/percona-server-mongodb-operator
+    tag: 1.4.0-backup
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/psmdb-db/production-values.yaml]: done

File: charts/psmdb-db/values.yaml (hunk 11 above)
Comment (cap1984) [reply charts/psmdb-db/values.yaml]: done

**Review by tplavcic — APPROVED** [review]

**Pull request merged**